File: frontend/src/types/activity.ts
export interface Activity {
  id: number
  title: string
  description: string
  city: string
  category: Category
  priceEur: number
  durationMinutes: number
  rating: number
  reviewCount: number
  imageUrl: string
  highlights: string[]
  isBookable: boolean
}

export enum Category {
  TOURS = 'TOURS',
  FOOD_AND_DRINK = 'FOOD_AND_DRINK',
  MUSEUMS = 'MUSEUMS',
  OUTDOOR = 'OUTDOOR',
  WORKSHOPS = 'WORKSHOPS',
  NIGHTLIFE = 'NIGHTLIFE',
  FAMILY = 'FAMILY',
  ADVENTURE = 'ADVENTURE',
}

export interface ChatMessage {
  role: 'user' | 'assistant'
  content: string
}

export interface ChatResponse {
  message: string
  recommendedActivityIds: number[]
}

File: backend/src/main/kotlin/com/gyg/concierge/model/Activity.kt
package com.gyg.concierge.model

data class Activity(
    val id: Long,
    val title: String,
    val description: String,
    val city: String,
    val category: Category,
    val priceEur: Double,
    val durationMinutes: Int,
    val rating: Double,
    val reviewCount: Int,
    val imageUrl: String,
    val highlights: List<String>,
    val isBookable: Boolean = true
)

enum class Category {
    TOURS,
    FOOD_AND_DRINK,
    MUSEUMS,
    OUTDOOR,
    WORKSHOPS,
    NIGHTLIFE,
    FAMILY,
    ADVENTURE
}
